splash/qtrender_lua.py

```diff
@@ -415,16 +416,39 @@ def error(error_info):
         return cmd
 
     @command(async=True)
-    def go(self, url, baseurl=None, headers=None):
+    def go(self, url, baseurl=None, headers=None, http_method="GET", body=None, formdata=None):
+
         if url is None:
             raise ScriptError({
                 "argument": "url",
                 "message": "'url' is required for splash:go",
             })
 
+        http_method = http_method.upper()
+        if http_method not in ["POST", "GET"]:
+            raise ScriptError("Unsupported HTTP method {}".format(http_method))
+
+        if formdata and body:
+            raise ScriptError("formdata and body cannot be passed to go() in one call")
+
+        elif formdata:
+            body = self.lua.lua2python(formdata, max_depth=3)
+            if isinstance(body, dict):
+                body = urlencode(body)
+            else:
+                raise ScriptError("formdata argument for go() must be Lua table")
+
+        elif body:
+            body = self.lua.lua2python(body, max_depth=3)
```

Comment:
well, someone could pass anything as body, if user passes Lua table or function or something else unexpected as body, it could cause trouble somewhere later, so I thought it's better raise explicit error here as soon as we know something is wrong. But to know what type we have we need to convert from lua to python.